src/avram/base_query_template.cr

```diff
@@ -112,13 +112,18 @@ class Avram::BaseQueryTemplate
             {% elsif assoc[:through] %}
               {{ join_type.downcase.id }}_join_{{ assoc[:through].id }}
                 .__yield_where_{{ assoc[:through].id }} do |join_query|
-                  join_query.{{ join_type.downcase.id }}_join_{{ assoc[:table_name] }}
+                  {% if assoc[:source].is_a?(NilLiteral) %}
+                    {% singular_association_name = run("../run_macros/singularize_word.cr", assoc[:assoc_name]) %}
+                    join_query.{{ join_type.downcase.id }}_join_{{ singular_association_name.id }}
+                  {% else %}
+                    join_query.{{ join_type.downcase.id }}_join_{{ assoc[:source].id }}
+                  {% end %}
```

Comment:
This is where we make the choice about the has_many through join. If a source is not specified we singularize the association name (we assume it is a belongs_to relationship)